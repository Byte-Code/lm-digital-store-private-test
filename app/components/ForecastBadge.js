import React from 'react';
import ImmutablePropTypes from 'react-immutable-proptypes';
import styled from 'styled-components';

import IconSelector from './Icons';

const Wrapper = styled.div`
  width: 235px;
  height: 280px;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
`;

const Day = styled.div`
  font-size: 24px;
  width: 100%;
  text-align: center;
  color: #fff;
  text-transform: uppercase;
`;

const Temp = styled.div`
  display: flex;
  flex-direction: column;
  &>p {
    text-align: center;
  }
`;

const GreenText = styled.span`
  font-size: 20px;
  text-transform: uppercase;
  color: #67cb33;
  margin-right: 10px;
`;

const WhiteText = styled.span`
  font-size: 40px;
  color: #fff;
`;

const IconWrapper = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
`;

const ForecastBadge = ({ dailyForecast }) => {
  const dt = dailyForecast.get('dt');
  const minTemp = dailyForecast.getIn(['temp', 'min']);
  const maxTemp = dailyForecast.getIn(['temp', 'max']);
  const weather = dailyForecast.getIn(['weather', 0, 'main']);

  return (
    <Wrapper>
      <Day>{dt}</Day>
      <IconWrapper>
        <IconSelector weather={weather} />
      </IconWrapper>
      <Temp>
        <p>
          <GreenText>min.</GreenText>
          <WhiteText>{minTemp} &#176;</WhiteText>
        </p>
        <p>
          <GreenText>max.</GreenText>
          <WhiteText>{maxTemp} &#176;</WhiteText>
        </p>
      </Temp>
    </Wrapper>
  );
};

ForecastBadge.propTypes = {
  dailyForecast: ImmutablePropTypes.map.isRequired
};

export default ForecastBadge;
