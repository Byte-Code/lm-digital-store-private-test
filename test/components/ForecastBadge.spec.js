import React from 'react';
import { fromJS } from 'immutable';
import { shallow } from 'enzyme';

import ForecastBadge from '../../app/components/ForecastBadge';

describe('ForecastBadge', () => {
  const dailyForecast = fromJS({
    dt: 'dt',
    temp: {
      min: 0,
      max: 0
    },
    weather: [{ main: 'sunny' }]
  });

  it('should render properly', () => {
    const result = shallow(<ForecastBadge dailyForecast={dailyForecast} />);
    expect(result).toMatchSnapshot();
  });
});
