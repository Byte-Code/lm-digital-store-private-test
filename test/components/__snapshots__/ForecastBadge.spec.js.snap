// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastBadge should render properly 1`] = `
<styled.div>
  <styled.div>
    dt
  </styled.div>
  <styled.div>
    <IconSelector
      weather="sunny"
    />
  </styled.div>
  <styled.div>
    <p>
      <styled.span>
        min.
      </styled.span>
      <styled.span>
        0
         °
      </styled.span>
    </p>
    <p>
      <styled.span>
        max.
      </styled.span>
      <styled.span>
        0
         °
      </styled.span>
    </p>
  </styled.div>
</styled.div>
`;
